kolibri-electron/src/index.js

```diff
@@ -10,14 +14,124 @@ if (require('electron-squirrel-startup')) { // eslint-disable-line global-requir
 }
 
 const KOLIBRI = 'http://localhost:5000';
-const pingTimeout = 20;
+let pingTimeout = 20;
+let mainWindow = null;
 let loadRetries = 0;
 let timeSpent = 0;
 let maxRetries = 3;
 
 let django = null;
 
-const waitForKolibriUp = (mainWindow) => {
+async function getEndlessKeyDrive() {
+  const drives = await drivelist.list();
+
+  // Look for the endless key drive
+  for (const d of drives.map((d) => d.mountpoints[0].path)) {
+    console.log(`Looking for endless Key on ${d}`);
+    const p = path.join(d, 'KOLIBRI_DATA');
+    try {
+      await fsPromises.access(p);
+      return d;
+    } catch (err) {
+      console.log(`${p} doesn't exists, trying next`);
+    }
+  };
+
+  return null;
+}
+
+async function getLoadingScreen() {
+  const defaultLoading = path.join(__dirname, 'Kolibri', 'assets', '_load.html');
+
+  // Try to use the custom loading screen from kolibri-explore-plugin
+  const drive = await getEndlessKeyDrive();
+  if (!drive) {
+    return defaultLoading;
+  }
+
+  const loading = path.join(
+    drive,
+    'KOLIBRI_DATA',
+    'extensions',
+    'kolibri_explore_plugin',
+    'loadingScreen',
+    'index.html',
+  );
+
+  try {
+    await fsPromises.access(loading);
+    return loading;
+  } catch (err) {
+    console.log(`Loading screen not found ${loading}`);
+  }
+
+  return defaultLoading;
+}
+
+async function getPluginVersion() {
+  const drive = await getEndlessKeyDrive();
+  if (!drive) {
+    console.error('Endless Key not found');
+    return '0.0.0';
+  }
+  const extensionsDir = path.join(drive, 'KOLIBRI_DATA', 'extensions');
+
+  try {
+    const files = await fsPromises.readdir(extensionsDir);
+    for (const file of files) {
+      // Looking for plugin kolibri_explore_plugin-VERSION.dist-info
+      const re = /kolibri_explore_plugin-(\d+\.\d+\.\d+)\.dist-info/
+      const match = file.match(re);
+      if (match) {
+        return match[1];
+      }
+    }
+  } catch (err) {
+    console.error(err);
+  }
+
+  return '0.0.0';
+}
+
+// Checks for the ~/.endless-key/version file and compares with the Endless key
+// kolibri-explore-plugin version. If the version is different, the
+// ~/endless-key folder will be removed.
+//
+// Return true if the version file does not match the plugin version or if the
+// .endless-key doesn't exists.
+async function checkVersion() {
+  console.log('Checking kolibri-app version file');
+  const kolibriHome = path.join(os.homedir(), '.endless-key');
+  const versionFile = path.join(kolibriHome, 'version');
+  const pluginVersion = await getPluginVersion();
+  let version = '0.0.0';
```

Comment:
This would be clearer if it's named something like `kolibriHomeVersion`.

Reply:
done